src/Common/Config/ConfigReloader.cpp

```diff
@@ -36,24 +36,31 @@ ConfigReloader::ConfigReloader(
 
 void ConfigReloader::start()
 {
-    thread = ThreadFromGlobalPool(&ConfigReloader::run, this);
+    std::lock_guard lock(reload_mutex);
+    if (!thread.joinable())
+    {
+        quit = false;
+        thread = ThreadFromGlobalPool(&ConfigReloader::run, this);
+    }
 }
 
 
-ConfigReloader::~ConfigReloader()
+void ConfigReloader::stop()
 {
-    try
-    {
-        quit = true;
-        zk_changed_event->set();
+    std::unique_lock lock(reload_mutex);
+    if (!thread.joinable())
+        return;
+    quit = true;
+    zk_changed_event->set();
```

Comment:
Why do we remove `try ... catch (...)` ? Can `zk_changed_event->set` throw an exception?

Reply:
I doubt it can throw exceptions but I've brought back `try ... catch (...)` here.